### PR TITLE
fix: bugfix for smartpay recovery flow

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-3259 MF-3237 MF-3267 MF-3232 MF-3148 MF-3084

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/dashboard/src/components/BackupPasswordConfirmDialog/index.tsx [hunk 1]
```diff
@@ -29,30 +30,47 @@ export const BackupPasswordConfirmDialog = memo<DialogProps>(({ onConfirmed, onC
         }
     }
 
+    const onSubmitPaymentPassword = useCallback(async () => {
+        const verified = await PluginServices.Wallet.verifyPassword(password)
```
Comment (guanbinrui) [inline packages/dashboard/src/components/BackupPasswordConfirmDialog/index.tsx:34]: Does it necessary use try catch here?

File: packages/mask/background/services/backup/restore.ts [hunk 2]
```diff
@@ -34,6 +44,69 @@ export async function restoreUnconfirmedBackup({ id, action }: RestoreUnconfirme
...
+}: {
+    mnemonic?: string
+    privateKeyString?: string
+    personaName: string
+}) {
+    let publicKey: EC_Public_JsonWebKey
+    let privateKey: EC_Private_JsonWebKey
+
+    if (mnemonic) {
+        const verify = bip39.validateMnemonic(mnemonic)
```
Comment (guanbinrui) [inline packages/mask/background/services/backup/restore.ts]: Rename it to verified.

File: packages/web3-hooks/base/src/useNonFungibleAssets.ts [hunk 3]
```diff
@@ -16,16 +16,18 @@ export function useNonFungibleAssets<S extends 'all' | void = void, T extends Ne
     const [done, setDone] = useState(false)
     const [loading, toggleLoading] = useState(false)
     const [error, setError] = useState<string>()
-
-    const { account } = useChainContext({ account: options?.account })
-    const hub = useWeb3Hub(pluginID)
+    const { Hub } = useWeb3State(pluginID)
+    const { account, chainId } = useChainContext({ account: options?.account })
+    // const hub = useWeb3Hub(pluginID)
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useNonFungibleAssets.ts]: Remove commented code.

File: packages/web3-hooks/base/src/useNonFungibleAssets.ts [hunk 4]
```diff
@@ -16,16 +16,18 @@ export function useNonFungibleAssets<S extends 'all' | void = void, T extends Ne
...
     const networks = useNetworkDescriptors(pluginID)
 
     // create iterator
     const iterator = useMemo(() => {
+        const hub = Hub?.getHub?.({ account, chainId, ...options })
         if (!account || !hub?.getNonFungibleAssets || !networks) return
         setAssets(EMPTY_LIST)
         setDone(false)
+        console.log(account)
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useNonFungibleAssets.ts]: Remove logs.

File: packages/dashboard/src/components/BackupPasswordConfirmDialog/index.tsx (hunk 1 above)
Comment (nuanyang233) [reply packages/dashboard/src/components/BackupPasswordConfirmDialog/index.tsx:34]: No need

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**